### ISSUE TITLE
Different behavior of %g format under Windows

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#6579]
**Original bug ID:** 6579
**Reporter:** @alainfrisch
**Status:** acknowledged (set by @damiendoligez on 2014-12-17T22:14:14Z)
**Resolution:** open
**Priority:** low
**Severity:** tweak
**Category:** platform support (windows, cross-compilation, etc)
**Tags:** junior_job
**Monitored by:** @gasche @hcarty

## Bug description

By default, Windows uses three digits for the exponent in scientific notation, while two digits are usually used on other systems.  This can give different result and break some non-regression tests that compare textual outputs.  I know there could be other sources of textual differences between Windows and Unix around display of floating point values, but this one seems to be easily fixable using  _set_output_format :

   http://msdn.microsoft.com/en-us/library/0fatw238%28v=vs.80%29.aspx

Is it a good idea to do that automatically upon startup of the OCaml runtime under Windows?

**Comment by vicuna** [comment]
**Comment author:** @gasche

I think that's a reasonable idea indeed.

**Comment by vicuna** [comment]
**Comment author:** @damiendoligez

Just a remark, this is a rare case where Windows gets it right and Unix gets it wrong: Unix adds some useless digits, but not enough to guarantee a fixed width.

I'm on the fence.

**Comment by jflopezfernandez** [comment]
Is this issue still active? I noticed it's from 2014.

I did some research and the `_set_output_format` function has actually [been deprecated](https://docs.microsoft.com/en-us/cpp/c-runtime-library/set-output-format?view=vs-2019), but as I was looking through the codebase I saw the `printf.ml` and `printf.mli` files, where they emulate the C library's own `printf` function, so maybe we could simply explicitly state the required padding with something like `%0.2f` or something like that? I'd have to see the tests that are failing, though, so I'm not sure if it's as quick a fix as that.
